microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -64,69 +65,69 @@ func getKubeConfig() (*rest.Config, error) {
 	return config, nil
 }
 
-// This is for linting purpose, they are supposed to be removed after reading manifests from oneclick
+// make go lint happy
 const (
-	Configmap         = "Configmap"
-	ConfigmapGaudi    = "ConfigmapGaudi"
-	Embedding         = "Embedding"
-	TeiEmbedding      = "TeiEmbedding"
-	TeiEmbeddingGaudi = "TeiEmbeddingGaudi"
-	VectorDB          = "VectorDB"
-	Retriever         = "Retriever"
-	Reranking         = "Reranking"
-	TeiReranking      = "TeiReranking"
-	Tgi               = "Tgi"
-	TgiGaudi          = "TgiGaudi"
-	Llm               = "Llm"
-	Router            = "router"
+	Configmap                        = "Configmap"
+	ConfigmapGaudi                   = "ConfigmapGaudi"
+	Embedding                        = "Embedding"
+	TeiEmbedding                     = "TeiEmbedding"
+	TeiEmbeddingGaudi                = "TeiEmbeddingGaudi"
+	VectorDB                         = "VectorDB"
+	Retriever                        = "Retriever"
+	Reranking                        = "Reranking"
+	TeiReranking                     = "TeiReranking"
+	Tgi                              = "Tgi"
+	TgiGaudi                         = "TgiGaudi"
+	Llm                              = "Llm"
+	Router                           = "router"
+	xeon                             = "xeon"
+	gaudi                            = "gaudi"
+	tei_reranking_service_yaml       = "/tei_reranking_service.yaml"
+	embedding_yaml                   = "/embedding.yaml"
+	tei_embedding_service_yaml       = "/tei_embedding_service.yaml"
+	tei_embedding_gaudi_service_yaml = "/tei_embedding_gaudi_service.yaml"
+	tgi_service_yaml                 = "/tgi_service.yaml"
+	tgi_gaudi_service_yaml           = "/tgi_gaudi_service.yaml"
+	llm_yaml                         = "/llm.yaml"
+	gmc_router_yaml                  = "/gmc-router.yaml"
+	redis_vector_db_yaml             = "/redis-vector-db.yaml"
+	retriever_yaml                   = "/retriever.yaml"
+	reranking_yaml                   = "/reranking.yaml"
 )
 
-func reconcileResource(step string, ns string, svc string, svcCfg *map[string]string, retSvc *corev1.Service) error {
+func reconcileResource(ctx context.Context, dynamicClient *dynamic.DynamicClient, step string, ns string, svc string, svcCfg *map[string]string, retSvc *corev1.Service) error {
 
 	var tmpltFile string
 
 	fmt.Printf("get step %s config for %s@%s: %v\n", step, svc, ns, svcCfg)
 
 	//TODO add validation to rule out unexpected case like both embedding and retrieving
-	if step == Configmap {
-		tmpltFile = yaml_dir + "/qna_configmap_xeon.yaml"
-	} else if step == ConfigmapGaudi {
-		tmpltFile = yaml_dir + "/qna_configmap_gaudi.yaml"
-	} else if step == Embedding {
-		tmpltFile = yaml_dir + "/embedding.yaml"
+	if step == Embedding {
+		tmpltFile = yaml_dir + embedding_yaml
 	} else if step == TeiEmbedding {
-		tmpltFile = yaml_dir + "/tei_embedding_service.yaml"
+		tmpltFile = yaml_dir + tei_embedding_service_yaml
```

Comment:
Would it have been easier to create a map {step  : yaml file}  tmpltFile = yaml_dir + yaml_files_map[step]?

Reply:
that's a great idea, thanks for comment